### PR TITLE
upgrade packages and use threadsjs

### LINKED ISSUE #40: update deps where possible
> It may be good to update packages listed in the package.json at some point, too
> eg: the latest version of @typescript-eslint/eslint-plugin is 4.28.1

from https://github.com/ChainSafe/simpleserialize.com/pull/39#pullrequestreview-698533223

### LINKED ISSUE #41: workerize-loader and webpack 5
The `workerize-loader` plugin that we are using does not support webpack 5 (pkg issue tracked here: https://github.com/developit/workerize-loader/issues/77).  So we have to use webpack 4 for now.  It also doesn't look like the package is being maintained much as this issue has been outstanding for a while despite there being 2 PRs in the package repo to solve the issue.

We could:
1) wait for the workerize-loader issue to be resolved
2) find a fork of workerize-loader that supports webpack 5
3) replace workerize-loader with something else that does the same thing
4) ???

### PR DESCRIPTION
**Pull request by 3xtr4t3rr3str14l**
Resolves #41 by upgrading to webpack 5 and using threadsjs for the web workers.

Since we were originally using the web-worker with an inline loader, but are no longer doing so under this PR, I also made a separate entry in the webpack config to transpile the worker `*.ts` file(s) into their own `worker.ts`

Moving from `workerize-loaders` to native web workers also opened the door to upgrade all the other packages in the repo, so I did that as well.

Also resolves #40

### CODE REVIEW

**Comment by 3xtr4t3rr3str14l** [comment]
I disabled `ban-types` since it was detected in the eslint upgrade.  I will make an issue to go through these instances, uncomment ban-types, and apply the changes required by the linter.

**Comment by dapplion** [comment]
@3xtr4t3rr3str14l Try this library, it's working great in Lodestar! https://www.npmjs.com/package/threads

**Comment by 3xtr4t3rr3str14l** [comment]
@dapplion I tried using threads, and I have most of the code setup, but I'm running into webpack/script loading/etc woes.  I've tried a bunch of things, and almost every time I try something I get these errors in the console:

```
GEThttp://localhost:8080/worker.tsx
[HTTP/1.1 404 Not Found 5ms]

Loading Worker from “http://localhost:8080/worker.tsx” was blocked because of a disallowed MIME type (“text/html”).
```

Any idea what might be going on?  I thought I had all the types right in my webpack config, but now I'm not sure.

**Comment by 3xtr4t3rr3str14l** [comment]
> @dapplion I tried using threads, and I have most of the code setup, but I'm running into webpack/script loading/etc woes. I've tried a bunch of things, and almost every time I try something I get these errors in the console:
> 
> ```
> GEThttp://localhost:8080/worker.tsx
> [HTTP/1.1 404 Not Found 5ms]
> 
> Loading Worker from “http://localhost:8080/worker.tsx” was blocked because of a disallowed MIME type (“text/html”).
> ```
> 
> Any idea what might be going on? I thought I had all the types right in my webpack config, but now I'm not sure. This wasn't happening when I was using the native web workers, fwiw.

fixed this.  ready for review again

File: src/components/Input.tsx [hunk 1]
```diff
@@ -32,29 +34,19 @@ function getRandomType(types: Record<string, Type<unknown>>): string {
...
 
-const workerInstance = worker();
-
 const DEFAULT_FORK = "phase0";
 
 class Input<T> extends React.Component<Props<T>, State> {
+  worker: Worker;
+  // TODO: not sure what type to put here
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  workerInstance: any;
```
Comment (wemeetagain) [inline src/components/Input.tsx:43]: This should be typed
https://threads.js.org/usage#type-safe-workers
You might also call it something more meaningful

File: src/components/worker/index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    const value = createRandomValue(type);
+    return value;
+  },
+  serialize(sszTypeName: string, forkName: string, input: object) {
+    const type = getSSZType(sszTypeName, forkName);
+    const serialized = type.serialize(input);
+    const root = type.hashTreeRoot(input);
+    return {serialized, root};
+  }
+}
```
Comment (wemeetagain) [inline src/components/worker/index.ts:18]: we don't also deserialize?

File: src/components/worker/index.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// /* eslint-disable @typescript-eslint/ban-types */
+
+import {expose} from "threads/worker";
+import {getSSZType, createRandomValue} from "./helpers";
+
+const worker = {
+  createRandomValueFromSSZType(sszTypeName: string, forkName: string) {
```
Comment (wemeetagain) [inline src/components/worker/index.ts:7]: Maybe can just be called `createRandomValue`

**Review by wemeetagain — CHANGES_REQUESTED** [review]

File: src/components/worker/index.ts (hunk 2 above)
Comment (3xtr4t3rr3str14l) [reply src/components/worker/index.ts:18]: actually, i don't think we are currently.  i'm going to make an issue to investigate and add the worker properly for deserialization.

**Review by dapplion — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**